File: C#/TopflytechCodec/TopflytechCodec/Entities/ObdMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TopflytechCodec.Entities
{
    class ObdMessage:Message
    {
        private String errorCode;

        public String ErrorCode
        {
            get { return errorCode; }
            set { errorCode = value; }
        }
        private String errorData;

        public String ErrorData
        {
            get { return errorData; }
            set { errorData = value; }
        }
        private DateTime date;

        public DateTime Date
        {
            get { return date; }
            set { date = value; }
        }


        private int messageType;

        public int MessageType
        {
            get { return messageType; }
            set { messageType = value; }
        }
        private String vin;

        public String Vin
        {
            get { return vin; }
            set { vin = value; }
        }
        private bool clearErrorCodeSuccess = false;

        public bool ClearErrorCodeSuccess
        {
            get { return clearErrorCodeSuccess; }
            set { clearErrorCodeSuccess = value; }
        }

        public static int ERROR_CODE_MESSAGE = 0;
        public static int VIN_MESSAGE = 1;
        public static int CLEAR_ERROR_CODE_MESSAGE = 2;
    }
}

File: Java/topflytechCodec/src/com/topflytech/codec/entities/ObdMessage.java
package com.topflytech.codec.entities;
 
import java.util.Date;

/**
 * Created by admin on 2017/10/20.
 */
public class ObdMessage extends Message{
    /**
     * Gets date.
     *
     * @return the date
     */
    public Date getDate() {
        return date;
    }

    /**
     * Sets date.
     *
     * @param date the date
     */
    public void setDate(Date date) {
        this.date = date;
    }



    public String getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(String errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorData() {
        return errorData;
    }

    public void setErrorData(String errorData) {
        this.errorData = errorData;
    }

    public int getMessageType() {
        return messageType;
    }

    public void setMessageType(int messageType) {
        this.messageType = messageType;
    }

    public String getVin() {
        return vin;
    }

    public void setVin(String vin) {
        this.vin = vin;
    }

    public boolean isClearErrorCodeSuccess() {
        return clearErrorCodeSuccess;
    }

    public void setClearErrorCodeSuccess(boolean clearErrorCodeSuccess) {
        this.clearErrorCodeSuccess = clearErrorCodeSuccess;
    }
    private String errorCode;
    private String errorData ;
    private Date date;


    private int messageType;
    private String vin;
    private boolean clearErrorCodeSuccess = false;

    public static int ERROR_CODE_MESSAGE = 0;
    public static int VIN_MESSAGE = 1;
    public static int CLEAR_ERROR_CODE_MESSAGE = 2;


}
